src/runtime/exceptions.cs

```diff
@@ -370,6 +370,10 @@ public static void deprecation(string message)
 
         internal static IntPtr RaiseTypeError(string message)
         {
+            if (Exceptions.ErrorOccurred())
+            {
+                message += Environment.NewLine + "  ----> " + new PythonException().Message;
+            }
```

Comment:
I think this should only be done explicitly. It is theoretically possible Python calls our code with the error flag already set, then it is not really a direct cause for the TypeError you are about to report.

Additionally, it is better to keep message fixed, and set `__cause__` (which, admittedly, I have not mastered yet).

Reply:
Is that really a problem though?  Extra information is always good for tracking down a bug.

Reply:
I have changed it to use __cause__.  If you don't want the existing error to be attached as the cause, then use SetError instead of RaiseTypeError.